Anbubin/4 2 24 release notes (#4192)
Azure/AKS: CHANGELOG.md

### Release Notes
* Features:
  * [AKS Cost Analysis](https://learn.microsoft.com/azure/aks/cost-analysis) is now generally available. View the aggregated costs for all your AKS clusters and namespaces in a subscription and drill into infrastructure and namespaces costs of a cluster directly in Azure Portal.
  * [Trusted Access on AKS cluster](https://learn.microsoft.com/en-us/azure/aks/trusted-access-feature) is generally available now.

* Preview Features:
  * [Disable SSH](https://learn.microsoft.com/en-us/azure/aks/manage-ssh-node-access?tabs=node-shell#disable-ssh-overview) is in preview now. Users can disable/enable the SSH access on nodepool level. 
  * Calico can now be disabled for an AKS cluster through the update operation. More info [here](https://learn.microsoft.com/en-us/azure/aks/use-network-policies#uninstall-azure-network-policy-manager-or-calico-preview).

* Behavioral Changes:

>   * We now allow customization of istiod and ingress gateway HPA to enable customers to ensure minreplicas does not go too low to break pod disruption budgets.

Reviewer: ```suggestion
  * Customizations to HorizontalPodAutoscaler (HPA) for `istiod` and Istio ingress gateways are now allowed. User can directly edit the HPAs in aks-istio-system and aks-istio-ingress namespaces to customize the HPA. Note that HPA changes that violate minReplicas specified in the existing PDB will be rejected/reset.
```

> * Bug Fixes: 

Reviewer 2: This is missing an entry for the following item:
>     Fix bug where the RP would sometimes normalize the case of networkProfile.loadBalancerSku from whatever case the user input it as such as 'standard' to 'Standard'.

See ADO PR: 9744704

Author: Any details on what issues/errors that caused?

Author: Fixed after chatting with Matthew async

# Azure Kubernetes Service Changelog

## Release 2024-03-31

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

### Announcements

> * Support upgrade version skew policy between core node and control plane components from n-2 to n-3 per [Kubernetes v1.28: Planternetes release note.](https://kubernetes.io/blog/2023/08/15/kubernetes-v1-28-release/) AKS docs available [here](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#alias-minor-version).

Reviewer: ```suggestion
* Support upgrade version skew policy between core node and control plane components from n-2 to n-3 to match [related upstream policy change starting Kubernetes version 1.28](https://kubernetes.io/blog/2023/08/15/kubernetes-v1-28-release/) AKS docs available [here](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#alias-minor-version).
```

> * Starting 1.30 kubernetes version and 1.27 LTS versions the beta apis will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/en-us/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to enable explicitly beta apis closer to the 1.30 release.

Reviewer: ```suggestion
* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta apis will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta apis closer to the 1.30 release.
```

> * If you use any programming or scripting logic to list and  select minor version before creating clusters - please ensure you update the logic to exclude LTS-only versions to avoid any breaks.  Otherwise, if LTS is your path forward please first opt into premium tier and LTS support plan as mentioned in [Long-term support for Azure Kubernetes Service (AKS) - Azure Kubernetes Service | Microsoft](https://learn.microsoft.com/en-us/azure/aks/long-term-support) before the create operation.

Reviewer: ```suggestion
* If you use any programming or scripting logic to list and  select minor version before creating clusters - please ensure you update the logic to exclude LTS-only versions to avoid any breaks.  Otherwise, if LTS is your path forward please first opt into premium tier and LTS support plan as mentioned in [Long-term support for Azure Kubernetes Service (AKS) - Azure Kubernetes Service | Microsoft](https://learn.microsoft.com/azure/aks/long-term-support) before the create operation.
```

* On 15 March 2027, Windows Server 2022 will be retired when Kubernetes 1.34 reaches the end of platform support. You won't be able to create new Windows Server 2022 node pools on Kubernetes 1.35 and above. We encourage you to make the switch before 15 March 2027 to gain the richer benefits of [Windows Server 2025](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/introducing-windows-server-2025/ba-p/4026374) or Windows Server [Annual Channel](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/windows-server-annual-channel-for-containers/ba-p/3866248). These new Windows OS versions will be supported on AKS before Windows Server 2022 is retired. For more updates, see our [AKS public roadmap](https://github.com/Azure/AKS/projects/1).

### Release Notes
* Features:
  * [AKS Cost Analysis](https://learn.microsoft.com/azure/aks/cost-analysis) is now generally available. View the aggregated costs for all your AKS clusters and namespaces in a subscription and drill into infrastructure and namespaces costs of a cluster directly in Azure Portal.

>   * [Trusted Access on AKS cluster](https://learn.microsoft.com/en-us/azure/aks/trusted-access-feature) is generally available now.

Reviewer: ```suggestion
  * [Trusted Access on AKS cluster](https://learn.microsoft.com/azure/aks/trusted-access-feature) is generally available now.
```

* Preview Features:

>   * [Disable SSH](https://learn.microsoft.com/en-us/azure/aks/manage-ssh-node-access?tabs=node-shell#disable-ssh-overview) is in preview now. Users can disable/enable the SSH access on nodepool level. 

Reviewer: ```suggestion
  * [Disable SSH](https://learn.microsoft.com/azure/aks/manage-ssh-node-access?tabs=node-shell#disable-ssh-overview) is in preview now. Users can disable/enable the SSH access on nodepool level. 
```

>   * Calico can now be disabled for an AKS cluster through the update operation. More info [here](https://learn.microsoft.com/en-us/azure/aks/use-network-policies#uninstall-azure-network-policy-manager-or-calico-preview).

Reviewer: ```suggestion
  * Calico can now be disabled for an AKS cluster through the update operation. More info [here](https://learn.microsoft.com//azure/aks/use-network-policies#uninstall-azure-network-policy-manager-or-calico-preview).
```

Author: Fixed

* Bug Fixes: 
  * Fixed missing CalicoBlockSize when uninstalling Calico. This fixes a bug that can cause the disablement of Calico Network Policies to fail.
  * Fixed an issue where node image upgrade or nodepool deletion might result in node auto provisioning to stop provisioning new nodes.
  * Fixed bug where the RP would sometimes normalize the case of networkProfile.loadBalancerSku from the case the user input, such as 'standard' to 'Standard', which may have caused diffs in Terraform state files or other client tools that perform diffs.

* Component Updates: 
  * Upgraded secrets-store-csi-driver to [version 1.4.2](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.4.2).
  * Upgraded azure-cloud-controller-manager to version [1.29.3](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.3), [1.28.8](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.8), [1.27.16](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.16), [1.26.22](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.26.22).

> 

Reviewer: we need to add vhd notes files and lines for those (Ubuntu, Windows, Azure Linux,...) here

Author: Added notes and files for ubuntu and linux, but windows was updated in last release notes (no change there).

# Azure Kubernetes Service Changelog

## Release 2024-03-31

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

### Announcements

> * Support upgrade version skew policy between core node and control plane components from n-2 to n-3 per [Kubernetes v1.28: Planternetes release note.](https://kubernetes.io/blog/2023/08/15/kubernetes-v1-28-release/)

Reviewer 3: On it i ll ensure we have a PR for it by EOD.

> * Support upgrade version skew policy between core node and control plane components from n-2 to n-3 per [Kubernetes v1.28: Planternetes release note.](https://kubernetes.io/blog/2023/08/15/kubernetes-v1-28-release/)

Author: Added MSFT doc link.

> * Starting 1.30 kubernetes version and 1.27 LTS versions the beta apis will be disabled by default. There will be an option provided  to enable explicitly beta apis closer to the 1.30 release. 

Reviewer 3: Yes can add this to the LTS page .

> * Starting 1.30 kubernetes version and 1.27 LTS versions the beta apis will be disabled by default. There will be an option provided  to enable explicitly beta apis closer to the 1.30 release. 

Reviewer 3: Starting 1.30 kubernetes version and 1.27 LTS versions the beta apis will be disabled by default, when you upgrade to [them.](https://learn.microsoft.com/en-us/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin) There will be an option provided  to enable explicitly beta apis closer to the 1.30 release.

> * Starting 1.30 kubernetes version and 1.27 LTS versions the beta apis will be disabled by default. There will be an option provided  to enable explicitly beta apis closer to the 1.30 release. 

Reviewer 3: @abubinski

> * Starting 1.30 kubernetes version and 1.27 LTS versions the beta apis will be disabled by default. There will be an option provided  to enable explicitly beta apis closer to the 1.30 release. 

Author: Updated, thanks!

> * If you use any programming/ scripting logic to list and  select minor version before creating clusters - please ensure you update the logic to exclude LTS-only versions to avoid any breaks.  Otherwise, if LTS is your path forward please first opt into premium tier and LTS support plan as mentioned in [Long-term support for Azure Kubernetes Service (AKS) - Azure Kubernetes Service | Microsoft](https://learn.microsoft.com/en-us/azure/aks/long-term-support) before the create operation.

Reviewer 3: Good point maybe a GH issue i ll work with @haitch

> * Calico can now be disabled for an AKS cluster through the update operation more info [here](https://learn.microsoft.com/en-us/azure/aks/use-network-policies#uninstall-azure-network-policy-manager-or-calico-preview).

Author: Fixed, thanks

### Release Notes
* Features:
  * [AKS Cost Analysis](https://learn.microsoft.com/azure/aks/cost-analysis) is now generally available. View the aggregated costs for all your AKS clusters and namespaces in a subscription and drill into infrastructure and namespaces costs of a cluster directly in Azure Portal.
  * [Trusted Access on AKS cluster](https://learn.microsoft.com/en-us/azure/aks/trusted-access-feature) is generally available now.

* Preview Features:
  * [Disable SSH](https://learn.microsoft.com/en-us/azure/aks/manage-ssh-node-access?tabs=node-shell#disable-ssh-overview) is in preview now. Users can disable/enable the SSH access on nodepool level. 

* Behavioral Changes:
  * We now allow customization of istiod and ingress gateway HPA to enable customers to ensure minreplicas does not go too low to break pod disruption budgets.

* Bug Fixes: 
  * Fixed AGIC crashing in USSec due to incorrect value for cloud environment. Changed the logic of determining "AZURECLOUDENVIRONMENT" variable in AGIC to be in line to other addons.

>   * Fixed missing CalicoBlockSize when uninstalling Calico. This fixes a bug that can cause uninstall of Calico to fail.

Reviewer 4: ... can cause the disablement of Calico Network Policies to fail.

Author: Fixed, thanks

>   * Fixed an issue where node image upgrade or agentpool deletion might result in Node Auto Provisioning stop provisioning new nodes.

Reviewer 4: nodepool deletion

...in node auto provisioning _to_ stop provisioning new nodes.

Author: Fixed, thanks

* Component Updates: 
  * Upgraded secrets-store-csi-driver to [version 1.4.2](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.4.2).
  * Upgraded azure-cloud-controller-manager to version [1.29.3](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.3), [1.28.8](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.8), [1.27.16](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.16), [1.26.22](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.26.22).

>   * Upgraded cloud-provider-azure to version [1.29.3](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.3), [1.28.8](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.8), [1.27.16](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.16), [1.26.22](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.26.22).

Reviewer 4: This lane and the one above seem duplicated given the links

Author: Fixed, thanks

  * Fixed bug where the RP would sometimes normalize the case of networkProfile.loadBalancerSku from the case the user input, such as 'standard' to 'Standard', which may have caused diffs in Terraform state files or other client tools that perform diffs.

* Component Updates: 
  * Upgraded secrets-store-csi-driver to [version 1.4.2](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.4.2).
  * Upgraded azure-cloud-controller-manager to version [1.29.3](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.3), [1.28.8](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.8), [1.27.16](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.16), [1.26.22](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.26.22).
  * Azure Linux image has been updated to [Azure Linux - 202403.25.0](vhd-notes/AzureLinux/202403.25.0.txt).
  * AKS Ubuntu 18.04 image has been updated to [AKSUbuntu-1804-202403.25.0](vhd-notes/aks-ubuntu/AKSUbuntu-1804/202403.25.0.txt).
  * AKS Ubuntu 22.04 image has been updated to [AKSUbuntu-2204-202403.25.0](vhd-notes/aks-ubuntu/AKSUbuntu-2204/202403.25.0.txt).

> 

Reviewer: we have Windows ones for this release too. need to copy notes from [17763.5576.240316](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2019-containerd/17763.5576.240316.txt)
for 2019 and
[20348.2340.240316](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2022-containerd-gen2/20348.2340.240316.txt)
for 2022

# Azure Kubernetes Service Changelog

## Release 2024-03-31

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

### Announcements
* Support upgrade version skew policy between core node and control plane components from n-2 to n-3 to match [related upstream policy change starting Kubernetes version 1.28](https://kubernetes.io/blog/2023/08/15/kubernetes-v1-28-release/) AKS docs available [here](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#what-is-the-allowed-difference-in-versions-between-control-plane-and-node-pool).
* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta apis will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta apis closer to the 1.30 release.

> * If you use any programming or scripting logic to list and  select minor version before creating clusters - please ensure you update the logic to exclude LTS-only versions to avoid any breaks.  Otherwise, if LTS is your path forward please first opt into premium tier and LTS support plan as mentioned in [Long-term support for Azure Kubernetes Service (AKS) - Azure Kubernetes Service | Microsoft](https://learn.microsoft.com/azure/aks/long-term-support) before the create operation.

Reviewer 4: Not sure if my previous comment landed or not, but this is still unclear on what is the bad case and what is the correct case customers should do. Also don't see a doc link that contains such info. @kaarthis @haitch

Reviewer 3: i ll start a GH issue first and we can clarify more there then follow up with a doc.  We can start populating from April rel. notes onward.  Suggest you can remove it from this one @abubinski

Author: Okay removed, thanks.

### Release Notes
* Features:
  * [AKS Cost Analysis](https://learn.microsoft.com/azure/aks/cost-analysis) is now generally available. View the aggregated costs for all your AKS clusters and namespaces in a subscription and drill into infrastructure and namespaces costs of a cluster directly in Azure Portal.
  * [Trusted Access on AKS cluster](https://learn.microsoft.com/en-us/azure/aks/trusted-access-feature) is GA now.

* Preview Features:
  * [Disable SSH](https://learn.microsoft.com/en-us/azure/aks/manage-ssh-node-access?tabs=node-shell#disable-ssh-overview) is in preview now. Users can disable/enable the SSH access on nodepool level. 

* Behavioral Changes:
  * We now allow customization of istiod and ingress gateway HPA to enable customers to ensure minreplicas does not go too low to break pod disruption budgets.

>   * Created ZRS disk by default in built-in storage classes on a multi-zone AKS cluster from AKS 1.29.

Reviewer 5: pls remove this since we found there is an issue, we will enable this behavior change in next aks rp release, thanks.

Author: Fixed, thanks!

# Azure Kubernetes Service Changelog

## Release 2024-03-31

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

> ### Announcements

Reviewer: we should carry over a few of the important announcements from previous release. for example Windows server retiring from K8s 1.34 one

Author: Copied that one over. Should I carryover all the others or something else specific?

> * Force migrating all Image Cleaner clusters to addon v2.

Author: Confirmed we can remove.

### Release Notes

> * Features:

Author: Added another for AKS Cost Analysis and will ask in Teams for anything else missing.

>   * Created ZRS disk by default in built-in storage classes on a multi-zone AKS cluster

Reviewer: Also is this GA or preview?

>   * Created ZRS disk by default in built-in storage classes on a multi-zone AKS cluster

Reviewer 5: it's a repair item instead of feature, we have internal postmortem icm to track this issue.

>   * Created ZRS disk by default in built-in storage classes on a multi-zone AKS cluster

Reviewer 5: Created ZRS disk by default in built-in storage classes on a multi-zone AKS cluster from AKS 1.29

>   * Created ZRS disk by default in built-in storage classes on a multi-zone AKS cluster

Author: I've updated and moved to behavior change since we don't have a specific section for repair items.

> * Component Updates: 

Reviewer 6: We generally write it as "Upgrade foo from vX.X.X to vY.Y.Y" and link each to change logs. 
Please review the past release notes: https://github.com/Azure/AKS/pull/4174/files

Author: Fixed, thank you

>   * Bump csi driver to v1.4.2.

Reviewer: which one? Files/disks/blob? And we need to include link to the release notes of each of those

Author: @andyzhangx , do you have the link for the release?

Reviewer 5: https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.4.2

Author: Thanks, updated!

>   * Bump cloud-controller-manager 1.29.3, 1.28.8, 1.27.16, 1.26.22

Reviewer: should be azure-cloud-controller-manager. and can we add link to the release notes for each of the versions. for 1.29.3 - https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.3

Author: Fixed, thank you

>   * Bump Windows image versions to 2024.3B Windows 2019: March 12, 2024—KB5035849 (OS Build 17763.5576). Version: 17763.5576.240316. Windows 2022: March 12, 2024—KB5035857 (OS Build 20348.2340). Gen1 Version: 20348.2340.240316. Gen2 Version: 20348.2340.240316.

Reviewer: vhd notes are missing. we need to add those and reference them similar to the previous release notes

Author: This was actually already announced in last release notes so I removed.

>   * Bump cloud-node-manager to 1.29.3, 1.28.8, 1.27.16, 1.26.22

Reviewer: same comment as the azure-cloud-controller-manager one

Author: I am not finding the link for this one - reached out to the engineer, but please send if you know it.

Reviewer 5: you could find ccm releases here: https://github.com/kubernetes-sigs/cloud-provider-azure/releases

>   * Bump agentbaker version to v0.20240325.0

Reviewer: we can skip this line. instead of this we need to check in the vhd notes for linux, windows,... and then we need to reference the same similar to previous release notes (for example, from the previous release note: Azure Linux image has been updated to [Azure Linux - 202401.17.0](https://github.com/Azure/AKS/blob/2024-01-23/vhd-notes/AzureLinux/202401.17.0.txt). )

Author: This was covered in last release notes, so I have removed.

  * Bump windows gpu device plugin image tag version to 0.0.3

>   * Support upgrade version skew policy between core node and control plane components from n-1 to n-3 per Kubernetes v1.28: Planternetes release note.

Reviewer: Need to move this under announcements section. Also, is the corresponding documentation update done?

Author: Moved and added release note link.

>   * EnableAdvancedNetworkingObservability helm value that enables Hubble on Cilium when advancedNetworking mc property is enabled and k8s version is 1.29. k8s 1.29 will always be cilium 1.14+, cilium versions are tied to which k8s version here: cilium-agent.yaml.

Reviewer: The helm value and cillium-agent.yaml are not directly customer facing. We need to reframe this to capture Hubble availability and the K8s version mappings

Author: I'm not getting a response on this, so I think it should be removed.

  * enable-azure-cni-overlay-dual-stack-windows allow-bypassing-k8s-dual-stack-upgrade-validation (this allows them to upgrade to dual-stack if they are < 1.28, but they will still need to be on k8s 1.26.3) allow-bypassing-cni-overlay-required-for-dual-stack-upgrade (this allows them to upgrade from cni v1 to overlay - we have an e2e test for this but it isn't in master validation)

>   * CCP plugin reconciliation was missing from the main loop. That already resulted in releases not being effective unless customers trigger PUT MC directly.

Reviewer: may not be customer facing. need to check internally

Author: Updated, thanks

>   * enable ccp-webhook-bind-to-all-with-tls toggle for all regions.

Reviewer: Not sure if this is customer facing, can we check internally

Author: Removed, thanks.

  * Previously, the cpu request is 400m, which is considered too high and may affect Pod's scheduling. This PR adjusts the cpu request to 100m, from observation, a normally running adapter only takes 10-30m CPU, so 100m should be a reasonable request.
  * Updating coredns to use image v1.9.4-hotfix.20240327 on all AKS clusters with version 1.24+. We still use coredns build version v1.9.4, we have only fixed the CVE vulnerabilities on this.
  * EnableAdvancedNetworkingObservability helm value that enables Hubble on Cilium when advancedNetworking mc property is enabled and k8s version is 1.29. k8s 1.29 will always be cilium 1.14+, cilium versions are tied to which k8s version here: cilium-agent.yaml.
  * Enable fast scaling on westus3. speed up scaling agentpool operation. scaling agentpool is the PUT AP request which only update agentpool count.
  * Return BadRequest when calling GetAgentPoolUpgradeProfileOperation for an agentpool using a customized image. Customized images are only used for internal testing. Some partner team may call it by mistakes. Reference: Incident 486354172 : prod eastus Frontend Qos failure for GetAgentPoolUpgradeProfileHandler.GET

> * Bug Fixes: 

Reviewer: We typically follow the following order under Release notes section - Features, Preview Features, Behavioral Change, Bug fixes, and Component Updates. Can we follow the same order for consistency

Author: Updated, thank you
